pkg/stanza/operator/parser/container/parser.go

```diff
@@ -0,0 +1,353 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package container // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/parser/json"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"sync"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+)
+
+const dockerFormat = "docker"
+const crioFormat = "crio"
+const containerdFormat = "containerd"
+const recombineInternalID = "recombine_container_internal"
+const dockerPattern = "^\\{"
+const crioPattern = "^(?P<time>[^ Z]+) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) ?(?P<log>.*)$"
+const containerdPattern = "^(?P<time>[^ ^Z]+Z) (?P<stream>stdout|stderr) (?P<logtag>[^ ]*) ?(?P<log>.*)$"
+const logpathPattern = "^.*\\/(?P<namespace>[^_]+)_(?P<pod_name>[^_]+)_(?P<uid>[a-f0-9\\-]+)\\/(?P<container_name>[^\\._]+)\\/(?P<restart_count>\\d+)\\.log$"
+const logPathField = "log.file.path"
+
+var (
+	logFieldsMapping = map[string]string{
+		"stream": "attributes.log.iostream",
+	}
+	k8sMetadataMapping = map[string]string{
+		"container_name": "attributes.k8s.container.name",
+		"namespace":      "attributes.k8s.namespace.name",
+		"pod_name":       "attributes.k8s.pod.name",
+		"restart_count":  "attributes.k8s.container.restart_count",
+		"uid":            "attributes.k8s.pod.uid",
+	}
+)
+
+// Parser is an operator that parses Container logs.
+type Parser struct {
```

Comment:
What do you think about having a single `logMatcher` or something like that instead of having all three of them?
In that case some code paths could be simplified and `detectFormat` could handle setting the appropriate one.

Reply:
We need 3 different regexp patterns to be compiled at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32594/files#diff-3fe6bc4c1fe7ae420dd94bf3780ccb5fe33ee6ed5e3746ed9fc0ee4f1c3b5c0eR65-R76.
I'm not sure how one `logMatcher` could cover those 3 different ones 🤔.